src/routes/getSegmentInfo.ts

```diff
@@ -34,11 +34,11 @@ async function handleGetSegmentInfo(req: Request, res: Response): Promise<DBSegm
     // deduplicate with set
     UUIDs = [ ...new Set(UUIDs)];
     // if more than 10 entries, slice
-    if (UUIDs.length > 10) UUIDs = UUIDs.slice(0, 10);
-    if (!Array.isArray(UUIDs) || !UUIDs) {
+    if (!Array.isArray(UUIDs) || !UUIDs?.length) {
         res.status(400).send("UUIDs parameter does not match format requirements.");
         return;
     }
+    if (UUIDs.length > 10) UUIDs = UUIDs.slice(0, 10);
```

Comment:
why?

Reply:
this way we can throw a proper 400 instead of 500 when TypeError occurs

Reply:
I'm confused, isn't this saying that if it is too long, we just ignore the extra stuff?

Reply:
This was part of the original segmentInfo, to only batch 10 UUIDs at at time, it was just moved to be **after** the isArray check instead

Reply:
for reference: https://discord.com/channels/603643120093233162/607338052221665320/854440486453510206